docker/docker_client.go

```diff
@@ -129,12 +138,29 @@ func newTransport() *http.Transport {
 	return tr
 }
 
+// dockerCertDir returns a path to a directory to be consumed by setupCertificates() depending on ctx and hostPort, or "" if no directory should be used.
+func dockerCertDir(ctx *types.SystemContext, hostPort string) string {
+	if ctx != nil && ctx.DockerCertPath != "" {
+		return ctx.DockerCertPath
+	}
+	var hostCertDir string
+	if ctx != nil && ctx.DockerPerHostCertDirPath != "" {
+		hostCertDir = ctx.DockerPerHostCertDirPath
+	} else if ctx != nil && ctx.RootForImplicitAbsolutePaths != "" {
+		hostCertDir = filepath.Join(ctx.RootForImplicitAbsolutePaths, systemPerHostCertDirPath)
+	} else {
+		hostCertDir = systemPerHostCertDirPath
```

Comment:
Does it make sense to add a log debug statement here?  I'm assuming this is something we don't want to happen often and it might be nice to easily debug if we need to.  Your call.